api/api/spec/spec.yaml

```diff
@@ -167,6 +167,14 @@ x-rbac-catalog:
         actions: ['agent:restart']
         resources: ['agent:id:050', 'agent:id:049']
         effect: "deny"
+    'agent:uninstall':
+      description: "Check user's permission to uninstall agents"
+      resources:
+        - $ref: '#/x-rbac-catalog/resources/*:*'
+      example:
+        actions: ['agent:uninstall']
+        resources: [ '*:*:*' ]
+        effect: "allow"
```

Comment:
Why is the action over every resource?

Reply:
@fdalmaup As described in https://github.com/wazuh/wazuh/issues/24369#issuecomment-2228769413, if the user has the action `agents:uninstall`, it can uninstall any agents. Due to this decision, the action is `resourceless`, so there is no associated resource to the action. All actions that are `resourceless` are associated with the resource `*:*:*`, for example, the `agent:create` and `agent:group`:

https://github.com/wazuh/wazuh/blob/ee9f897c02fee8c8469abefc1e792785deace598/framework/wazuh/rbac/default/policies.yaml#L4-L14

In this case, `agent:create` can create new agents or can't, there is no in-between, equal to `agent:uninstall` and doesn't make any changes/modifications to the already created resources in the system, so it doesn't need to have a specific resource.

If we add. the resource `agent:id:*` to `agent:uninstall` we end with the second approach described in https://github.com/wazuh/wazuh/issues/24369#issuecomment-2223365815.